src/it/java/teammates/it/storage/sqlapi/FeedbackResponseCommentsDbIT.java

```diff
@@ -45,4 +56,133 @@ public void testGetFeedbackResponseCommentForResponseFromParticipant() {
 
         assertEquals(expectedComment, actualComment);
     }
+
+    @Test
+    public void testUpdateFeedbackResponseComment_noChangeToComment_shouldNotIssueSaveRequest() throws Exception {
+        FeedbackResponseComment frc = typicalDataBundle.feedbackResponseComments.get("comment1ToResponse1ForQ1");
+
+        ______TS("OPTIMIZED_SAVING_POLICY_APPLIED should trigger on original");
+        // please verify the log message manually to ensure that saving request is not issued
+        FeedbackResponseComment updatedComment = frcDb.updateFeedbackResponseComment(frc);
+        assertEquals(JsonUtils.toJson(frc), JsonUtils.toJson(updatedComment));
+
+        updatedComment = new FeedbackResponseComment(
+            frc.getFeedbackResponse(),
+            frc.getGiver(),
+            frc.getGiverType(),
+            frc.getGiverSection(),
+            frc.getRecipientSection(),
+            frc.getCommentText(),
+            frc.getIsVisibilityFollowingFeedbackQuestion(),
+            frc.getIsCommentFromFeedbackParticipant(),
+            frc.getShowCommentTo(),
+            frc.getShowGiverNameTo(),
+            frc.getLastEditorEmail()
+        );
+        updatedComment.setId(frc.getId());
```

Comment:
If I am not wrong, I think the ID should not change. If it changes, it might be a bug. I realise now that there are also a few other places where it is assumed they can be updated; I think they should be removed.

Reply:
Yup, I missed this out, thanks

Reply:
Shifted to https://github.com/TEAMMATES/teammates/pull/12745